CRAMv3.tex

```diff
@@ -1539,31 +1542,80 @@ \subsubsection*{General notes}
+The exception to this is with multi-reference containers, where the ``RI'' data series must be read.
 
 \subsubsection*{CRAM index}
 
 A CRAM index is a gzipped tab delimited file containing the following columns:
 
 \begin{enumerate}
-\item Sequence id
+\item Reference sequence id
 
 \item Alignment start (ignored on read for unmapped slices, set to 0 on write)
 
 \item Alignment span (ignored on read for unmapped slices, set to 0 on write)
 
-\item Container start byte offset in the file
+\item Absolute byte offset of Container structure in the file.
 
-\item Slice start byte offset in the container data (`blocks')
+\item Relative byte offset of the Slice header block, from the end of
+the container structure.  This is the same as the ``landmark'' field
+in the container structure.
 
-\item Slice bytes
+\item Slice size in bytes (including slice header and all blocks).
 \end{enumerate}
 
-Each line represents a slice in the CRAM file. Please note that all slices must 
-be listed in the index file.
+Each line represents a slice in the CRAM file.
+Please note that all slices must be listed in the index file.
+
+Multi-reference slices may need to have multiple lines for the same slice; one for each reference contained within that slice.
+In this case the index reference sequence ID will be the actual reference number (from the ``RI'' data series) and not -2.
+
+Slices containing solely unmapped unplaced data (reference ID -1) still require values for all columns, although the alignment start and span will be ignored.
+It is recommended that they are set to small constant values (0 or 1).
+
+To illustrate this the absolute and relative offsets used in a three slice container are shown in the diagram below.
+
+\begin{tikzpicture}
+% 0   -  2.5
+% 2.5 -  5.5
+% 5.5 -  7.5
+% 7.5 -  9.5
+% 7.5 - 11.5
+\draw [black] (0,0) rectangle (2.5,1.5);
+\node at (1,0.75) [text width=1cm]{Container Header};
+
+\draw [black] (2.5,0) rectangle (5.5,1.5);
+\node at (3.5,0.75) [text width=1cm]{Compression Header Block};
+
+\draw [black] (5.5,0) rectangle (7.5,1.5);
+\node at (6.5,0.75) [text width=1cm]{Slice 1};
+
+\draw [black] (7.5,0) rectangle (9.5,1.5);
+\node at (8.5,0.75) [text width=1cm]{Slice 2};
+
+\draw [black] (9.5,0) rectangle (11.5,1.5);
+\node at (10.5,0.75) [text width=1cm]{Slice 3};
+
+\draw[|<->|] (2.5,-.3) -- (5.5,-.3);
+\node [below] at (4,-0.3) {Slice offset 1};
+
+\draw[|<->|] (2.5,-1.3) -- (7.5,-1.3);
+\node [below] at (4,-1.3) {Slice offset 2};
+
+\draw[|<->|] (2.5,-2.3) -- (9.5,-2.3);
+\node [below] at (4,-2.3) {Slice offset 3};
+
+\draw[<-] (0,-0.1) -- (0,-1) node[below, text width=1.5cm]{Container absolute offset};
+
+\draw[|<->] (5.5,1.8) -- (7.5,1.8);
+\draw[|<->] (7.5,1.8) -- (9.5,1.8);
+\draw[|<->|] (9.5,1.8) -- (11.5,1.8);
+\node [above] at (6.5,1.8) {Size 1};
+\node [above] at (8.5,1.8) {Size 2};
+\node [above] at (10.5,1.8){Size 3};
```

Comment:
Nice!

Reply:
It's my first time exploring pgf/tikz.  I already see I can write this in a friendlier fashion, but it's a good start and I like the ability to version control our diagrams.

Reply:
Yes indeed :smile:

Trivially, perhaps wrap `\begin{center}` around it? That appears to be the way to centre a `tikzpicture`…

Reply:
Now centred, and rewritten the pic layout.

Useful resources I found:

http://www-groups.mcs.st-andrews.ac.uk/~alanc/pub/c_tikzref/c_tikzref.pdf

https://stuff.mit.edu/afs/athena/contrib/tex-contrib/beamer/pgf-1.01/doc/generic/pgf/version-for-tex4ht/en/pgfmanualse8.html